### ISSUE TITLE
Support heredoc `<<-` redirection operator 

### ISSUE THREAD

**Issue by Massedil** [de-jcup/eclipse-bash-editor#169]
For better indenting the code, it is possible to use the `<<-` redirection operator.

```
If the redirection operator is ‘<<-’, then all leading tab characters are stripped from input lines and the line containing delimiter. This allows here-documents within shell scripts to be indented in a natural fashion. 
```

Example :

```
if true ; then
    cat <<- EOF > /tmp/file
    plop
    EOF
fi
```

https://www.gnu.org/software/bash/manual/bash.html#Here-Documents

Now, if this `<<-` redirection operator is used in a bash script :
- syntax coloration is wrong
- wrong errors are reported

It could also be interesting in that case to show a difference between `whitespace` and `tab` because `whitespace` will be kept and `tab` will be ignored.

**Comment by de-jcup** [comment]
Hello, I tried out your example code, with bash editor version 2.2.0 and I do not have any problems:

![image](https://user-images.githubusercontent.com/11644753/60401464-23c9eb00-9b82-11e9-8bb8-b5d5a8b78a92.png)

- no syntax coloring problem
- no errors shown

Could you upload/provide a complete example file which has such problem?

**Comment by Massedil** [comment]
> Could you upload/provide a complete example file which has such problem?

Sorry, after a better look, in fact, it is when you have both `<<-`  and a quote `'` inside the text that the parsing problem occurs.

```
#!/bin/bash
if true ; then
    cat <<- EOF > /tmp/file
	Lorem Ipsum is simply dummy text
	of the printing and typesetting industry. 
	Lorem Ipsum has been the industry's standard 
	it to make a type specimen book. 
	essentially unchanged. It was popula
	EOF
fi

```

![](https://user-images.githubusercontent.com/20357783/60438660-89b18380-9c11-11e9-97a0-9a16479fe8ad.png)

**Comment by de-jcup** [comment]
I started the script above with `GNU bash, version 4.4.19(1)-release (x86_64-pc-linux-gnu)` and got following error message:

```
albert@devtux:~/testscripts/features/feature-169$ ./feature-169-heredoc-redirection2.sh 
./feature-169-heredoc-redirection2.sh: line 10: warning: here-document at line 3 delimited by end-of-file (wanted `EOF')
./feature-169-heredoc-redirection2.sh: line 11: syntax error: unexpected end of file
```

so script seems to be not correct

**Comment by Massedil** [comment]
Here a valid [test.txt](https://github.com/de-jcup/eclipse-bash-editor/files/3349389/test.txt) file (because github refuse test.sh) and commands I have run.

- Can you download the file and try it ?

```
# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 8.11 (jessie)
Release:        8.11
Codename:       jessie

# bash --version
GNU bash, version 4.3.30(1)-release (x86_64-pc-linux-gnu)

# cat /root/test.txt
#!/bin/bash
if true ; then
        cat <<- EOF > /tmp/file
           Lorem Ipsum is simply dummy text
           of the printing and typesetting industry.
           Lorem Ipsum has been the industry's standard
           it to make a type specimen book.
           essentially unchanged. It was popula
        EOF
fi

# bash /root/test.txt

# cat /tmp/file
   Lorem Ipsum is simply dummy text
   of the printing and typesetting industry.
   Lorem Ipsum has been the industry's standard
   it to make a type specimen book.
   essentially unchanged. It was popula
```

- Can you reproduce my test ?

![](https://user-images.githubusercontent.com/20357783/60503124-59282300-9cbf-11e9-8d9b-ae809e1da38f.png)

**Comment by de-jcup** [comment]
Hello @Massedil , yes it works now and I can reproduce your  test.

**Comment by de-jcup** [comment]
Is now fixed in parser and also in UI rendering:
![image](https://user-images.githubusercontent.com/11644753/60546414-87d4e680-9d1d-11e9-991f-8df0eabab360.png)

**Comment by de-jcup** [comment]
Just released V2.2.1 on eclipse marketplace containing the bugfix. Thanx for reporting.

**Comment by Massedil** [comment]
>Just released V2.2.1 on eclipse marketplace containing the bugfix. Thanx for reporting.

Thank you !
